src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -1057,6 +1061,44 @@ void ZooKeeper::pushRequest(RequestInfo && info)
     ProfileEvents::increment(ProfileEvents::ZooKeeperTransactions);
 }
 
+KeeperApiVersion ZooKeeper::getApiVersion()
+{
+    return keeper_api_version;
+}
+
+void ZooKeeper::initApiVersion()
+{
+    auto promise = std::make_shared<std::promise<Coordination::GetResponse>>();
+    auto future = promise->get_future();
+
+    auto callback = [promise](const Coordination::GetResponse & response) mutable
+    {
+        promise->set_value(response);
+    };
+
+    get(keeper_api_version_path, std::move(callback), {});
+    if (future.wait_for(std::chrono::milliseconds(operation_timeout.totalMilliseconds())) != std::future_status::ready)
+    {
+        LOG_TRACE(&Poco::Logger::get("ZooKeeper"), "Failed to get API version: timeout");
```

Comment:
```suggestion
        LOG_TRACE(log, "Failed to get API version: timeout");
```
https://github.com/ClickHouse/ClickHouse/blob/1d26446f8413ec9821e9e97961786e04570755c3/src/Common/ZooKeeper/ZooKeeperImpl.cpp#L321